File: 18 09 25/lengthlast.js
// LeetCode 58: Length of Last Word
// Explanation:
// 1. trim() removes trailing/leading spaces
// 2. Iterate from end to count last word length
// Time Complexity: O(n)

var lengthOfLastWord = function(s) {
    s = s.trim();
    let i = s.length - 1, length = 0;
    while (i >= 0 && s[i] !== ' ') {
        length++;
        i--;
    }
    return length;
};

// Example usage:
// console.log(lengthOfLastWord("Hello World"));            // 5
// console.log(lengthOfLastWord("   fly me   to   the moon  ")); // 4
// console.log(lengthOfLastWord("a"));                      // 1

File: 18 09 25/lengthlast.py
# LeetCode 58: Length of Last Word
# Explanation:
# 1. strip() removes trailing/leading spaces.
# 2. split() breaks the string into words by spaces.
# 3. words[-1] gives the last word.
# 4. len() returns its length.
# Time Complexity: O(n) where n = length of s

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        words = s.strip().split()
        return len(words[-1])
    
